#==================================================================================================
# Upstream patch, which fixes compilation issues due to conflicting OpenGL-related headers.
#
# https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/6208
#==================================================================================================

--- gdk/macos/gdkmacosglcontext-private.h
+++ gdk/macos/gdkmacosglcontext-private.h
@@ -27,9 +27,7 @@
 #include "gdkmacosdisplay.h"
 #include "gdkmacossurface.h"
 
-#import <OpenGL/OpenGL.h>
-#import <OpenGL/gl3.h>
-#import <AppKit/AppKit.h>
+#import <epoxy/gl.h>
 
 G_BEGIN_DECLS
 
diff --git a/gdk/macos/gdkmacosglcontext.c b/gdk/macos/gdkmacosglcontext.c
index 87e6e19fe0a..a706a407b5d 100644
--- gdk/macos/gdkmacosglcontext.c
+++ gdk/macos/gdkmacosglcontext.c
@@ -21,7 +21,6 @@
 
 #include "gdkconfig.h"
 
-#include <OpenGL/gl3.h>
 #include <OpenGL/CGLIOSurface.h>
 #include <QuartzCore/QuartzCore.h>
 
@@ -499,7 +498,7 @@ gdk_macos_gl_context_begin_frame (GdkDrawContext *context,
   gdk_gl_context_make_current (GDK_GL_CONTEXT (self));
   gdk_macos_gl_context_allocate (self);
 
-  GDK_DRAW_CONTEXT_CLASS (gdk_macos_gl_context_parent_class)->begin_frame (context, prefers_high_depth, region);
+  GDK_DRAW_CONTEXT_CLASS (gdk_macos_gl_context_parent_class)->begin_frame (context, depth, region);
 
   gdk_gl_context_make_current (GDK_GL_CONTEXT (self));
   CHECK_GL (NULL, glBindFramebuffer (GL_FRAMEBUFFER, self->fbo));
